### PR TITLE
Test: When adding a new member to academy, if it exists already as member with role=student it should update it instead of creating it. #1504

### PR DESCRIPTION
**Pull request by 99lalo**

### CODE REVIEW

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 1]
```diff
@@ -469,3 +469,155 @@ def test_academy_id_member_with_two_roles(self):
...
+            'first_name': None,
+            'id': 1,
+            'last_name': None,
+            'phone': '',
+            'role_id': role,
+            'status': 'INVITED',
+            'user_id': 2,
+        }])
+
+    def test_academy_id_member_post_user_with_student_role(self):
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:589]: We need a new test like it, the current profile as student and the new as teacher

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 2]
```diff
@@ -469,3 +469,179 @@ def test_academy_id_member_with_two_roles(self):
...
         } for index in range(0, 2)])
+
+    def test_academy_id_member_post_no_data(self):
+        """Test /academy/:id/member"""
+        role = 'konan'
+        model = self.generate_models(authenticate=True, role=role,
+            capability='crud_member', profile_academy=True)
+        url = reverse_lazy('authenticate:academy_id_member', kwargs={'academy_id':1})
+        data = {}
+        response = self.client.post(url, data)
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:480]: Test the data in `response.json()`

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 3]
```diff
@@ -469,3 +469,179 @@ def test_academy_id_member_with_two_roles(self):
...
+    def test_academy_id_member_post_no_user(self):
+        """Test /academy/:id/member"""
+        role = 'konan'
+        model = self.generate_models(authenticate=True, role=role,
+            capability='crud_member', profile_academy=True)
+        url = reverse_lazy('authenticate:academy_id_member', kwargs={'academy_id':1})
+        data = {'role': role}
+        response = self.client.post(url, data)
+        json = response.json()
+        expected = {'detail': "User does not exists, do you want to invite it?",
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:505]: Use the `slug` of `ValidationException`

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 4]
```diff
@@ -469,3 +469,181 @@ def test_academy_id_member_with_two_roles(self):
...
+            'status': 'ACTIVE',
+            'user_id': 2,
+        }])
+
+    def test_academy_id_member_post_teacher_with_student_role(self):
+        """Test /academy/:id/member"""
+        role = 'student'
+        model = self.generate_models(authenticate=True, role=role,
+            capability='crud_member', profile_academy=True)
+        model2 = self.generate_models(role='teacher', capability='crud_member')
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:621]: I think this line should be removed

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 5]
```diff
@@ -469,3 +469,181 @@ def test_academy_id_member_with_two_roles(self):
...
+    def test_academy_id_member_post_no_invite(self):
+        """Test /academy/:id/member"""
+        role = 'konan'
+        model = self.generate_models(authenticate=True, role=role,
+            capability='crud_member', profile_academy=True)
+        url = reverse_lazy('authenticate:academy_id_member', kwargs={'academy_id':1})
+        data = {'role': role, 'invite': True}
+        response = self.client.post(url, data)
+        json = response.json()
+        expected = {'detail': "Please specify user id or member email",
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:535]: Add slug

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py [hunk 6]
```diff
@@ -469,3 +469,181 @@ def test_academy_id_member_with_two_roles(self):
...
+    def test_academy_id_member_post_user_with_not_student_role(self):
+        """Test /academy/:id/member"""
+        role = 'konan'
+        model = self.generate_models(authenticate=True, role=role,
+            capability='crud_member', profile_academy=True)
+        url = reverse_lazy('authenticate:academy_id_member', kwargs={'academy_id':1})
+        data = { 'role': role , 'user' : model['user'].id, 'first_name': 'Kenny', 'last_name': 'McKornick'}
+        response = self.client.post(url, data)
+        json = response.json()
+        expected = {'detail': f'This user is already a member of this academy as {role} ({role})',
```
Comment (jefer94) [inline breathecode/authenticate/tests/urls/tests_academy_id_member.py:563]: Add slug

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/authenticate/tests/urls/tests_academy_id_member.py (hunk 4 above)
Comment (jefer94) [reply breathecode/authenticate/tests/urls/tests_academy_id_member.py:621]: The model2

**Review by jefer94 — APPROVED** [review]

**Comment by jefer94** [comment]
@alesanchezr is done

**Comment by jefer94** [comment]
https://github.com/4GeeksAcademy/About-4Geeks-Academy/issues/1504

**Review by alesanchezr — APPROVED** [review]

**Pull request merged**